mylyn.tasks/connectors/gitlab/org.eclipse.mylyn.gitlab.core/src/org/eclipse/mylyn/gitlab/core/GitlabRestClient.java

```diff
@@ -0,0 +1,989 @@
+/*******************************************************************************
+ * Copyright (c) 2023 Frank Becker and others.
+ * 
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0
+ * 
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Frank Becker - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.mylyn.gitlab.core;
```

Comment:
We need to put more work here before promoting this class to API, I suggest to move to `internal` package until it will be ready.

Reply:
Could you use the openapi spec  (https://gitlab.com/gitlab-org/gitlab/-/blob/master/doc/api/openapi/openapi.yaml?plain=1) to generate the rest client code?

Reply:
I will look into this in the near future how useful the generated client can be.
We can not generate the whole client because have hat to support TaskData Structure but hopefully this is only a small change.

Reply:
> We need to put more work here before promoting this class to API, I suggest to move to `internal` package until it will be ready.

done with next commit